src/frame.rs

```diff
@@ -60,7 +60,8 @@ impl PhysFrame {
 
 impl Drop for PhysFrame {
     fn drop(&mut self) {
-        if self.start_paddr.as_usize() > 0 {
+        // Do not deallocate the frame if it is a empty frame.
+        if self.start_paddr.as_usize() > 0 && self.start_paddr.as_usize() != 0xdead_beef {
```

Comment:
What is the magic number here? Explanation needed.

Reply:
I thought `0xdead_beef` is a well known magic number, it will be claimed as a `const PADDR_UNINIT: usize = 0xdead_beef` with detailed comments.

Reply:
> I thought `0xdead_beef` is a well known magic number, it will be claimed as a `const PADDR_UNINIT: usize = 0xdead_beef` with detailed comments.

When and where will it be assigned `0xdeadbeef`?

Reply:
here: https://github.com/arceos-hypervisor/x86_vcpu/blob/9e3621db5d02a62defdd15287a59286ece7c38ce/src/frame.rs#L44

Reply:
or maybe just set it as zero if more general?

I just thought that set it as 0 may cause some implicit bugs

Reply:
`Option` is better.

Reply:
> `Option` is better.


I think so too.